tests/pds/pds_restore_negative_test.go

```diff
@@ -0,0 +1,350 @@
+						log.FailOnError(err, "Failed while fetching pod for stateful set %v.", deployment.GetClusterResourceName())
+						err = sourceTarget.KillPodsInNamespace(params.InfraToTest.Namespace, podName)
+						log.FailOnError(err, "Failed while deleting pod.")
+						err = dsTest.ValidateDataServiceDeployment(deployment, nsName)
+						log.FailOnError(err, "Failed while validating the deployment pods, post pod deletion.")
+					}
+
+				})
+				stepLog = "Perform adhoc backup and validate them"
+				Step(stepLog, func() {
+					log.InfoD(stepLog)
+					dsEntity = restoreBkp.DSEntity{
+						Deployment: deployment,
+					}
+					log.Infof("Deployment ID: %v, backup target ID: %v", deployment.GetId(), bkpTarget.GetId())
+					err = bkpClient.TriggerAndValidateAdhocBackup(deployment.GetId(), bkpTarget.GetId(), "s3")
+					log.FailOnError(err, "Failed while performing adhoc backup")
+				})
+
+				stepLog = "Perform restore for the backup jobs."
+				Step(stepLog, func() {
+					log.InfoD(stepLog)
+					restoreClient := restoreBkp.RestoreClient{
+						TenantId:             tenantID,
+						ProjectId:            projectID,
+						Components:           components,
+						Deployment:           deployment,
+						RestoreTargetCluster: restoreTargetCluster,
+					}
+					backupJobs, err := restoreClient.Components.BackupJob.ListBackupJobsBelongToDeployment(projectID, deployment.GetId())
+					log.FailOnError(err, "Error while fetching the backup jobs for the deployment: %v", deployment.GetClusterResourceName())
+					for _, backupJob := range backupJobs {
+						log.Infof("[Restoring] Details Backup job name- %v, Id- %v", backupJob.GetName(), backupJob.GetId())
+						restoredModel, err := restoreClient.TriggerAndValidateRestore(backupJob.GetId(), params.InfraToTest.Namespace, dsEntity, true, true)
+						log.FailOnError(err, "Failed during restore.")
+						restoredDeployment, err = restoreClient.Components.DataServiceDeployment.GetDeployment(restoredModel.GetDeploymentId())
+						log.FailOnError(err, fmt.Sprintf("Failed while fetching the restore data service instance: %v", restoredModel.GetClusterResourceName()))
+						deploymentsToBeCleaned = append(deploymentsToBeCleaned, restoredDeployment)
+						log.InfoD("Restored successfully. Deployment- %v", restoredModel.GetClusterResourceName())
+					}
+				})
+
+				// TODO trigger workload for restored deployment
+
+				Step("Delete Deployments", func() {
+					CleanupDeployments(deploymentsToBeCleaned)
+				})
+			}
+		})
+	})
+	JustAfterEach(func() {
+		defer EndTorpedoTest()
+		defer CleanupDeployments(deploymentsToBeCleaned)
+		err := bkpClient.AWSStorageClient.DeleteBucket()
+		log.FailOnError(err, "Failed while deleting the bucket")
+	})
+})
+
+var _ = Describe("{PerformRestorePDSPodsDown}", func() {
+	bkpTargetName = bkpTargetName + pdsbkp.RandString(8)
+	JustBeforeEach(func() {
+		StartTorpedoTest("PerformRestoreToSameCluster", "Perform restore while validating HA.", pdsLabels, 0)
```

Comment:
Can we have the description more specific to the testcase scenario.

Reply:
Done